### PR TITLE
Split up View type to improve TS linting

### PR DESCRIPTION
**Pull request by alexjamesmalcolm**
###  Summary

There a `View` type. It has three subset types `modal`, `home`, and `workflow_step`. As I've discovered while consuming `@slack/bolt` there is a benefit to split these three up. Specifically that the `modal` subset requires the `title` field but the other two subsets have no use for it. If a consumer of this attempts to send a `modal` view without a `title` they will receive an error so by splitting up the single `interface` into three we can prevent the consumer from having to make the same mistake as myself to learn that although the `title` field is optional for a `view`, when `type: "modal"` is the case the `title` field is required.

I had to rely on these two pieces of documentation to get all of the fields correct:
* [`workflow_step` docs](https://api.slack.com/reference/workflows/configuration-view)
* [`home` and `modal` docs](https://api.slack.com/reference/surfaces/views)

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/types/src/index.ts
Comment (seratch) [inline packages/types/src/index.ts:53]: The API document may not be clear enough but as of today, this option is available only for "workflow_step" type modals. Can you remove `submit_disabled` from this type definition? 
```suggestion
```

**Review by seratch — CHANGES_REQUESTED** [review]
Thanks for the great pull request! Can you check my comment?

**Review by seratch — APPROVED** [review]
LGTM! Before merging this, I would like to wait for others' comments for a few days

**Review by misscoded — APPROVED** [review]

**Review by srajiang — APPROVED** [review]

**Comment by srajiang** [comment]
I think we should be good to merge these changes in, unless there was further discussion to be had on this @seratch!

**Comment by seratch** [comment]
Thanks @srajiang, yes, we can merge this PR 👍

**Pull request merged**
